gui/notes/index.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @module notes
+ * @flow
+ */
+
+const { models } = require('cozy-client')
+
+const path = require('path')
+const metadata = require('../../core/metadata')
+const {
+  CozyDocumentMissingError,
+  CozyMissingPermissionsError,
+  UnreachableError
+} = require('../../core/remote/errors')
+
+const logger = require('../../core/utils/logger')
+const log = logger({
+  component: 'Notes'
+})
+
+/*::
+import { Shell } from 'electron'
+import { App } from '../../core/app'
+import { Config } from '../../core/config'
+import { Pouch } from '../../core/pouch'
+import { Remote } from '../../core/remote'
+import type { Metadata } from '../../core/metadata'
+
+type CozyFile = Object
+*/
+
+/**
+ * computeNoteURL - Constructs a link to the Note within the web app
+ *
+ * @param {object} Options               Object of options
+ * @param {string}   options.noteId        Id of the CouchDB document representing the given note
+ * @param {string}   options.cozyUrl       Base URL of the cozy, eg. cozy.tools or test.mycozy.cloud
+ * @param {string}   options.subDomainType Whether the cozy is using flat or nested subdomains. Defaults to flat.
+ *
+ * @return {string} Generated URL
+ */
+const computeNoteURL = async ({ noteId, cozyUrl, subDomainType }) => {
+  const slug = 'notes'
+  const url = new URL(cozyUrl)
+  const nativePath = `/n/${noteId}`
+  url.host =
+    subDomainType === 'nested'
+      ? `${slug}.${url.host}`
+      : url.host
+          .split('.')
+          .map((x, i) => (i === 0 ? x + '-' + slug : x))
+          .join('.')
+  url.hash = nativePath
+  return url.toString()
+}
+
+const options = async (
+  noteId /*: string */,
+  { config, remote } /*: { config: Config, remote: Remote } */
+) => {
+  return {
+    noteId,
+    cozyUrl: config.cozyUrl,
+    subDomainType: (await remote.usesFlatDomains()) ? 'flat' : 'nested'
+  }
+}
+
+const localDoc = async (
+  filePath /*: string */,
+  { config, pouch } /*: { config: Config, pouch: Pouch } */
+) => {
+  const relPath = path.relative(config.syncPath, filePath)
+  const doc = await pouch.byIdMaybeAsync(metadata.id(relPath))
+  if (!doc) {
+    throw new CozyDocumentMissingError({
+      cozyURL: config.cozyUrl,
+      doc: { name: path.basename(relPath) }
+    })
+  }
+  return doc
+}
+
+const remoteDoc = async (
+  localDoc /*: Metadata */,
+  { config, remote } /*: { config: Config, remote: Remote } */
+) /*: CozyFile */ => {
+  const doc = await remote.remoteCozy.find(localDoc.remote._id)
+  if (!doc) {
+    throw new CozyDocumentMissingError({
+      cozyURL: config.cozyUrl,
+      doc: { name: path.basename(localDoc.path) }
+    })
+  }
+  return doc
+}
+
+const isMyNote = (
+  note /* CozyFile */,
+  { config } /*: { config: Config } */
+) => {
+  const isNote = models.file.isNote(note)
+  const isMine = note.cozyMetadata.createdOn.startsWith(config.cozyUrl)
+
+  return isNote && isMine
+}
```

Comment:
Maybe we can add some tests for the functions in this file

Reply:
We can also extract the isMine method to `models.file.isMine` since it'll also be shared 👍

Reply:
@Crash-- that's a good idea. I'll make a PR.